yjit/bindgen/src/main.rs

```diff
@@ -0,0 +1,286 @@
+        .blocklist_type("rb_execution_context_.*") // Large struct with various-type fields and an ifdef, so we don't import
+        .opaque_type("rb_execution_context_.*")
+        .blocklist_type("rb_control_frame_struct")
+        .opaque_type("rb_control_frame_struct")
+
+        // From yjit.c
+        .allowlist_function("rb_iseq_(get|set)_yjit_payload")
+        .allowlist_function("rb_iseq_pc_at_idx")
+        .allowlist_function("rb_iseq_opcode_at_pc")
+        .allowlist_function("rb_yjit_mark_writable")
+        .allowlist_function("rb_yjit_mark_executable")
+        .allowlist_function("rb_yjit_get_page_size")
+        .allowlist_function("rb_leaf_invokebuiltin_iseq_p")
+        .allowlist_function("rb_leaf_builtin_function")
+        .allowlist_function("rb_set_cfp_(pc|sp)")
+        .allowlist_function("rb_cfp_get_iseq")
+        .allowlist_function("rb_yjit_multi_ractor_p")
+        .allowlist_function("rb_c_method_tracing_currently_enabled")
+        .allowlist_function("rb_full_cfunc_return")
+        .allowlist_function("rb_yjit_vm_lock_then_barrier")
+        .allowlist_function("rb_yjit_vm_unlock")
+        .allowlist_function("rb_assert_(iseq|cme)_handle")
+        .allowlist_function("rb_IMEMO_TYPE_P")
+        .allowlist_function("rb_iseq_reset_jit_func")
+        .allowlist_function("rb_yjit_dump_iseq_loc")
+        .allowlist_function("rb_yjit_for_each_iseq")
+        .allowlist_function("rb_yjit_obj_written")
+
+        // from vm_sync.h
+        .allowlist_function("rb_vm_barrier")
+
+        // Not sure why it's picking these up, but don't.
+        .blocklist_type("FILE")
+        .blocklist_type("_IO_.*")
+
+        // From internal/compile.h
+        .allowlist_function("rb_vm_insn_decode")
+
+        // From iseq.h
+        .allowlist_function("rb_vm_insn_addr2opcode")
+        .allowlist_function("rb_iseqw_to_iseq")
+        .allowlist_function("rb_iseq_each")
+
+        // From builtin.h
+        .allowlist_type("rb_builtin_function.*")
+
+        // From internal/variable.h
+        .allowlist_function("rb_gvar_(get|set)")
+        .allowlist_function("rb_obj_ensure_iv_index_mapping")
+
+        // From include/ruby/internal/intern/variable.h
+        .allowlist_function("rb_attr_get")
+        .allowlist_function("rb_ivar_get")
+
+        // From include/ruby/internal/intern/vm.h
+        .allowlist_function("rb_get_alloc_func")
+
+        // From gc.h and internal/gc.h
+        .allowlist_function("rb_class_allocate_instance")
+        .allowlist_function("rb_obj_info")
+
+        // We define VALUE manually, don't import it
+        .blocklist_type("VALUE")
+
+        // From iseq.h
+        .opaque_type("rb_iseq_t")
+        .blocklist_type("rb_iseq_t")
+
+        // Finish the builder and generate the bindings.
+        .generate()
+        // Unwrap the Result and panic on failure.
+        .expect("Unable to generate bindings");
+
+    let mut out_path: PathBuf = src_root;
+    out_path.push("yjit");
```

Comment:
That's not true, extend() for PathBuf is equivalent to `for x in it { p.push(x) }`, but you *can* call push() with a path that has separators in it. `out_path.push("yjit/src/cruby_bindings.inc.rs")` is fine and cross-platform, and it can actually just be `let out_path = src_root.join("yjit/src/cruby_bindings.inc.rs");` at that point.

Reply:
Ayy the current code looks fine to me. We don't need this to be fast and it looks similar to what I find in the docs.